crates/ruff_workspace/src/configuration.rs

```diff
@@ -603,6 +628,18 @@ impl Configuration {
             );
         }
 
+        for selection in deprecated_nursery_selectors {
+            let (prefix, code) = selection.prefix_and_code();
+            warn_user!("Selection of nursery rule `{prefix}{code}` without the `--preview` flag is deprecated.",);
```

Comment:
Because this doesn't use `warn_user_once_by_id`, the warnings will probably show up multiple times if we run this configuration resolution multiple times (which we might in a given invocation -- I can't remember exactly). Is it possible to use `warn_user_once_by_id`?

Reply:
Yeah but I'd have to construct an id string like "nursery-{prefix}{code}", is that okay? I was thinking of adding a `warn_user_once_by_message` utility as an alternative? 🤷‍♀️

Reply:
I think that's okay

Reply:
But I think the ID has to be a static string so that may not work...